### DISCUSSION TITLE
Cách gõ giống Unikey

### DISCUSSION THREAD

**Discussion by hukadev** [BambooEngine/ibus-bamboo#287] [General]
Chào mọi người,

Hiện mình đang dùng 2 OS cả Windows và Linux. Và có gặp một vấn đề là cách gõ để ra chữ nó không giống nhau giữa ibus-bamboo và Unikey.
Ví dụ gõ Network ở Unikey thì sẽ ra Network, còn ibus-bamboo sẽ ra Netwỏk (nên cần phải gõ Networrk để ra Network)

Nếu được thì nhờ mọi người tạo bài viết sơ sơ cách contribute project này, một vài method, func cần chú ý được không nhỉ?

**Comment by t1ld3x** [comment]
Hi, bạn kiểm tra đã bật tính năng `Kiểm tra chính tả` chưa rồi thử lại nhé. 
Về cách contribute thì khá dễ, 99% logic của bộ gõ nằm ở hàm `ProcessKeyEvent` https://github.com/BambooEngine/ibus-bamboo/blob/f6406437969b8e40c5e529cad0055d894fad2c7a/src/ibus-bamboo/engine.go#L77  rất đơn giản 😄. 
Lưu ý ibus-bamboo sử dụng nhiều cơ chế gõ khác nhau (7 chế độ gõ) và một số trick/hack cho một vài ứng dụng (thanh địa chỉ trình duyệt, GG sheets, facebook,...) nên cần test càng nhiều ứng dụng càng tốt trước khi tạo merge request. Nếu có unit test hoặc automation test thì tuyệt vời 💯
